gitlab.go

```diff
@@ -505,8 +516,20 @@ func newClient(httpClient *http.Client) *Client {
 		RetryMax:     30,
 	}
 
-	// Set the default base URL.
-	_ = c.SetBaseURL(defaultBaseURL)
+	// Set the default base URL and apply any given client options.
+	for _, fn := range append([]ClientOptionFunc{WithBaseURL(defaultBaseURL)}, options...) {
+		if fn == nil {
```

Comment:
I'm wondering what is the reason to allow for this?

Reply:
This is actually not needed anymore as it's solved in a different way now. So this one is removed 👍

Reply:
Sorry, it's still here. Just reminding in case you wanted to remove the `if` but forgot.